src/crypto/crypto_tls.cc

```diff
@@ -144,32 +150,47 @@ int NewSessionCallback(SSL* s, SSL_SESSION* sess) {
     return 0;
 
   // Serialize session
-  // TODO(@jasnell): An AllocatedBuffer or BackingStore would be better
-  // here to start eliminating unnecessary uses of Buffer where an ordinary
-  // Uint8Array would do just fine.
-  Local<Object> session = Buffer::New(env, size).FromMaybe(Local<Object>());
-  if (UNLIKELY(session.IsEmpty()))
+
+  Local<Object> buffer_prototype_object = env->buffer_prototype_object();
+  if (UNLIKELY(buffer_prototype_object.IsEmpty()))
     return 0;
 
-  unsigned char* session_data =
-      reinterpret_cast<unsigned char*>(Buffer::Data(session));
+  Local<Object> session;
+  {
+    std::unique_ptr<BackingStore> bs =
+        ArrayBuffer::NewBackingStore(isolate, size);
+
+    unsigned char* session_data = reinterpret_cast<unsigned char*>(bs->Data());
+    i2d_SSL_SESSION(sess, &session_data);
+
+    Local<ArrayBuffer> ab = ArrayBuffer::New(isolate, std::move(bs));
+    Local<Uint8Array> ui = Uint8Array::New(ab, 0, ab->ByteLength());
+    if (UNLIKELY(ui->SetPrototype(context,
+                                  buffer_prototype_object).IsNothing()))
+      return 0;
 
-  memset(session_data, 0, size);
```

Comment:
I decided to remove this as there was no requirement of filling `pp` with `0`s in https://www.openssl.org/docs/man1.0.2/man3/i2d_SSL_SESSION.html

Reply:
I feel like it should be okay if (and only if) you also `CHECK` that the return value of `i2d_SSL_SESSION` is `size`.

Reply:
@addaleax Updated to use a `CHECK` but I was wondering why it wouldn't be okay to just gracefully `return 0;` in case the return value is `0` or not `size`. I'm asking because that's the pattern that's being followed inside this function.

Reply:
No idea tbh, I just care about not leaking uninitialized memory here. My guess is that the return value will always be `size` anyway (hence my suggestion to add a `CHECK`), so that’s fine with me. :woman_shrugging: